File: Contracts/Model/Product/Response/OptionsResponse.cs
namespace Contracts.Model.Product.Response;
public class OptionsResponse
{
	public int Id { get; set; }
	public string Title { get; set; }
	public string? Description { get; set; }
	public double Price { get; set; }
	public string? Image { get; set; }
	public string Stock => StockCalculate().ToString();
	public int QuantityAvailable { get; set; }
	public string Condition { get; set; }
	private StockStatusType StockCalculate()
	{
		if (QuantityAvailable == 0)
			return StockStatusType.OutOfStock;
		if (QuantityAvailable > 0 && QuantityAvailable < 10)
			return StockStatusType.LowStock;
		if (QuantityAvailable >= 10)
			return StockStatusType.InStock;
		return StockStatusType.InStock;
	}
}

File: ClientApp/src/app/core/interfaces/Product/Response/optionsResponse.ts
export interface OptionsResponse {
  id: string;
  title: string;
  description: string | null;
  price: number;
  color: string | null;
  genderFor: number;
  condition: number;
  stock: string;
  quantityAvailable: number;
  images: any[];
  imageId: number | null;
}
